plugins/language_haxe.lua

```diff
@@ -0,0 +1,111 @@
+-- mod-version:3
+local syntax = require "core.syntax"
+
+syntax.add {
+  name = "Haxe Compiler Arguments",
+  files = "%.hxml$",
+  patterns = {
+    { pattern = "%-[%-%w_]*",      type="keyword"},
+    { pattern = "%.()%u[%w_]*",    type = {"normal", "keyword2"}},
+  },
+  symbols = {}
+}
+
+syntax.add {
+  name = "Haxe String Interpolation",
+  files = "%.hx__string_interp$",
+  patterns = {
+    { pattern = {"%${", "}", "\\"},     type="keyword", syntax = ".hx" },
+    { pattern = {"%$", "%s", "\\"},     type="keyword", syntax = ".hx" },
+    { pattern = "[^ ]",                 type = "string"}
+  },
+  symbols = {}
+}
+
+syntax.add {
+  name = "Haxe Regular Expressions",
+  files = "%.hx__regex$",
+  patterns = {
+    { pattern = "[%[%]%(%)]",           type = "string" },
+    { pattern = "[%.%*%+%?%^%$%|%-]",   type = "operator" },
+  },
+  symbols = {}
+}
+
+syntax.add {
+  name = "Haxe",
+  files = "%.hx$",
+  comment = "//",
+  patterns = {
+    { pattern = {"%~%/", "%/[igmsu]*"},                     type = "keyword2", syntax = ".hx__regex" },
+    { pattern = "%.%.%.",                                   type = "operator" },
+    { pattern = "%<()%u[%w_]*()%>*",                        type = {"operator", "keyword2", "operator"}},
+    { pattern = "%#%s*[%a_]*().*\n",                        type = {"keyword", "normal"}   },
+    { pattern = "import%s+()%u[%w]*",                       type = {"keyword2", "keyword2"}},
+    { pattern = "import%s+()[%w%.]*%.()%u[%w]*",            type = {"keyword2", "normal", "keyword2"}},
+    { pattern = "abstract%s+()%u[%w_]*%s*%(()%s*%u[%w_]*",  type = {"keyword2", "normal", "keyword2"} },
+    { pattern = "from%s+()%u[%w_]*%s+()to%s+()%u[%w_]*",    type = {"keyword", "keyword2", "keyword", "keyword2"}},
+    { pattern = "//.*\n",                                   type = "comment"  },
+    { pattern = { "/%*", "%*/" },                           type = "comment"  },
+    { pattern = { '"', '"', '\\' },                         type = "string"   },
+    { pattern = { "'", "'", "\\" },                         type = "string", syntax = ".hx__string_interp"},  
+    { pattern = "-?%.?%d+",                                 type = "number"   },
+    { pattern = "-?0x%x+",                                  type = "number"   },
+    { pattern = "-?%d+%.[%deE]+",                           type = "number"   },
+    { pattern = "-?%d+[%deE]+",                             type = "number"   },
+    { pattern = "[%+%-%.=/%*%^%%<>!~|&]",                   type = "operator" },
+    { pattern = "[%a_][%w_]*%s*%f[(]",                      type = "function" },
```

Comment:
This one seems more difficult to deal with. That change fixes the `if (` problem, but breaks highlighting for function calls. I looked at the C++ highlighter to compare, and it seem to have a similar issue, though it opts to break highlighting for `func (` rather than `if (`. I think that's better since it's less common to see the former than the latter.

See the attached screenshots for comparison:

Proposed change:
![haxe_1](https://user-images.githubusercontent.com/656330/219417464-bfc4f661-0b63-43f6-86db-521c12a6bed0.png)

New version similar to language_cpp:
![haxe_2](https://user-images.githubusercontent.com/656330/219417472-c1a05da9-02fb-46b5-b9b4-b0d005b1f3ce.png)

Unfortunately, it doesn't seem like lite xl gives precedence to keywords when highlighting, so this is probably as good as it'll get for now. (tested on master)